test/test_torch.py

```diff
@@ -1142,6 +1142,36 @@ def test_pdist_norm(self):
                         self.assertEqual(expected.shape, actual.shape)
                         self.assertTrue(torch.allclose(expected, actual))
 
+
+    def test_cdist_empty(self):
+        devices = ['cpu'] if not torch.cuda.is_available() else ['cpu', 'cuda']
+        for device in devices:
+            for r1 in [0, 2]:
+                for m in[0, 5]:
+                    for r2 in[0, 3]:
+                        x = torch.zeros((r1, m), device=device)
+                        y = torch.zeros((r2, m), device=device)
+                        if r1 == 0 or r2 == 0:
+                            self.assertEqual(torch.empty(r1, r2, device=device), torch.cdist(x, y))
+                        else:
+                            self.assertEqual(torch.zeros(r1, r2, device=device), torch.cdist(x, y))
+
+    def test_cdist_norm(self):
+        devices = ['cpu'] if not torch.cuda.is_available() else ['cpu', 'cuda']
+        for device in devices:
+            for r1 in [3, 4, 5, 6]:
+                for m in [2, 3, 4, 10]:
+                    for r2 in [4, 6, 7, 8]:
+                        for p in [0, 1, 2, 3, 1.5, 2.5, float('inf')]:
+                            x = torch.randn(r1, m, device=device)
+                            y = torch.randn(r2, m, device=device)
+                            actual = torch.cdist(x, y, p=p)
+                            expected = brute_cdist(x, y, p=p)
+                            self.assertEqual(expected.shape, actual.shape)
+                            self.assertTrue(torch.allclose(expected, actual))
+
+
```

Comment:
as above, add at least a couple of large inputs (you don't have to do it for all values of p).